cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -689,24 +695,48 @@ protected Future<Void> brokerConfigurationConfigMaps() {
-                                    kafka.generateCertificatesSecret(clusterCa, clientsCa, listenerReconciliationResults.bootstrapDnsNames, listenerReconciliationResults.brokerDnsNames, Util.isMaintenanceTimeWindowsSatisfied(reconciliation, maintenanceWindows, clock.instant())))
-                            .compose(patchResult -> {
-                                if (patchResult != null) {
-                                    for (NodeRef node : kafka.nodes()) {
-                                        kafkaServerCertificateHash.put(
-                                                node.nodeId(),
-                                                CertUtils.getCertificateThumbprint(patchResult.resource(),
-                                                        Ca.SecretEntry.CRT.asKey(node.podName())
-                                                ));
-                                    }
+                .compose(oldSecret -> secretOperator
+                        .reconcile(reconciliation, reconciliation.namespace(), KafkaResources.kafkaSecretName(reconciliation.name()),
+                                kafka.generateCertificatesSecret(clusterCa, clientsCa, listenerReconciliationResults.bootstrapDnsNames, listenerReconciliationResults.brokerDnsNames, Util.isMaintenanceTimeWindowsSatisfied(reconciliation, maintenanceWindows, clock.instant())))
+                        .compose(patchResult -> {
+                            if (patchResult != null) {
+                                for (NodeRef node : kafka.nodes()) {
+                                    Secret secret = patchResult.resource();
+                                    String crtKey = Ca.SecretEntry.CRT.asKey(node.podName());
+
+                                    kafkaServerCertificateHash.put(
+                                            node.nodeId(),
+                                            CertUtils.getCertificateThumbprint(secret, crtKey));
+
+                                    emitCertificateSecretMetrics(patchResult.getType(), secret, crtKey);
                                 }
+                            }
 
-                                return Future.succeededFuture();
-                            });
-                });
+                            return Future.succeededFuture();
+                        }));
+    }
+
+    /**
+     * Emits the certificate expiration metric for the given secret if the resultType is CREATED or PATCHED,
+     * if DELETED the metric will be set to 0 for resetting.
+     *
+     * @param resultType Type of the ReconcileResult
+     * @param secret The modified secret
+     * @param crtKey The crt key for the modified secret
+     */
+    private void emitCertificateSecretMetrics(ReconcileResult.Type resultType, Secret secret, String crtKey) {
+        switch (resultType) {
+            case CREATED, PATCHED, NOOP:
+                metrics.certificateExpiration(kafka.getCluster(), reconciliation.namespace()).set(
```

Comment:
Should we expose the metric per node? At some points, f.e on cert update state might be inconsistent between brokers

Reply:
@scholzj and @ppatierno what do you think about @maciej-tatarskis comment - whats the needs of https://github.com/strimzi/strimzi-kafka-operator/issues/3761? :)